sentry-android-replay/src/main/java/io/sentry/android/replay/ScreenshotRecorder.kt

```diff
@@ -122,21 +126,42 @@ internal class ScreenshotRecorder(
                             )
                             val canvas = Canvas(scaledBitmap)
                             canvas.setMatrix(prescaledMatrix)
-                            viewHierarchy.traverse {
-                                if (it.shouldRedact && (it.width > 0 && it.height > 0)) {
-                                    it.visibleRect ?: return@traverse
-
-                                    // TODO: check for view type rather than rely on absence of dominantColor here
-                                    val color = if (it.dominantColor == null) {
-                                        singlePixelBitmapCanvas.drawBitmap(bitmap, it.visibleRect, Rect(0, 0, 1, 1), null)
-                                        singlePixelBitmap.getPixel(0, 0)
-                                    } else {
-                                        it.dominantColor
+                            viewHierarchy.traverse { node ->
+                                if (node.shouldRedact && (node.width > 0 && node.height > 0)) {
+                                    node.visibleRect ?: return@traverse false
+
+                                    if (viewHierarchy.isObscured(node)) {
```

Comment:
since you need to traverse every node here, this is a n^2 operation. Not saying that this is bad, but we should probably profile this at some point.

Reply:
yep, for sure. I don't think there's any other way though, at least not from the top of my head. I've tried to optimize it with DFS and skipping traversal of irrelevant nodes/subtrees, so n^2 is really the worst case here, and probably will never happen, because the likelihood of all the nodes' visible rects intersecting is very small